src/operations/indexes.ts

```diff
@@ -51,14 +51,17 @@ const VALID_INDEX_OPTIONS = new Set([
 
 /** @public */
 export type IndexDirection = -1 | 1 | '2d' | '2dsphere' | 'text' | 'geoHaystack' | number;
-
+function isIndexDirection(x: any): x is IndexDirection {
+  return (
+    typeof x === 'number' || x === '2d' || x === '2dsphere' || x === 'text' || x === 'geoHaystack'
+  );
+}
 /** @public */
 export type IndexSpecification = OneOrMore<
   | string
   | [string, IndexDirection]
   | { [key: string]: IndexDirection }
-  | [string, IndexDirection][]
-  | { [key: string]: IndexDirection }[]
+  | Map<string, IndexDirection>
```

Comment:
@dariakp @baileympearson @durran Perhaps we omit the Map as a possible value from here and in `IndexDescription`. This just adds an inconsistency we haven't done elsewhere in our APIs I'm thinking we might want to consider a better approach if we want our APIs to effectively communicate what it supported. For example, if `Document` was actually separated `InputDocument` and `OutputDocument` the idea would be that everywhere where the driver accepts some kind of BSON object it could actually accept:
```ts
type InputDocument = Record<string, any> | Map<string, any>
```
This small example obviously has big limitations but perhaps there is some better driver wide solution to the fact that Maps can be accepted anywhere BSON documents are taken. Adding this Map here now could prevent us from nicely solving this in the future, (ex. the Document type doesn't have assertions on the value type but the Map does). Thoughts?